src/selectors/ProfilePageSelector.js

```diff
@@ -97,24 +83,231 @@ export const editableFormSelector = createSelector(
+          visibilityName: preferences.visibilityName || 'private',
+          visibilitySocialLinks: preferences.visibilitySocialLinks || 'private',
+        };
+      case 'private':
+      default:
+        // If there is some other value for accountPrivacy set, we're going to assume
+        // it's private, since that's safest.
+        return {
+          visibilityBio: 'private',
+          visibilityCourseCertificates: 'private',
+          visibilityCountry: 'private',
+          visibilityEducation: 'private',
+          visibilityLanguageProficiencies: 'private',
+          visibilityName: 'private',
+          visibilitySocialLinks: 'private',
+        };
+    }
+  },
+);
+
+/**
+ * If there's no draft present at all (undefined), use the original committed value.
+ */
+function chooseFormValue(draft, committed) {
+  return draft !== undefined ? draft : committed;
+}
+
+export const formValuesSelector = createSelector(
+  profileAccountSelector,
+  visibilitiesSelector,
+  profileDraftsSelector,
+  profileCourseCertificatesSelector,
+  formSocialLinksSelector,
+  (account, visibilities, drafts, courseCertificates, socialLinks) => ({
+    bio: chooseFormValue(drafts.bio, account.bio),
+    visibilityBio: chooseFormValue(drafts.visibilityBio, visibilities.visibilityBio),
+    courseCertificates,
+    visibilityCourseCertificates: chooseFormValue(
+      drafts.visibilityCourseCertificates,
+      visibilities.visibilityCourseCertificates,
+    ),
+    country: chooseFormValue(drafts.country, account.country),
+    visibilityCountry: chooseFormValue(drafts.visibilityCountry, visibilities.visibilityCountry),
+    education: chooseFormValue(drafts.education, account.education),
+    visibilityEducation: chooseFormValue(
+      drafts.visibilityEducation,
+      visibilities.visibilityEducation,
+    ),
+    languageProficiencies: chooseFormValue(
+      drafts.languageProficiencies,
+      account.languageProficiencies,
+    ),
+    visibilityLanguageProficiencies: chooseFormValue(
+      drafts.visibilityLanguageProficiencies,
+      visibilities.visibilityLanguageProficiencies,
+    ),
+    name: chooseFormValue(drafts.name, account.name),
+    visibilityName: chooseFormValue(drafts.visibilityName, visibilities.visibilityName),
+    socialLinks, // Social links is calculated in its own selector, since it's complicated.
+    visibilitySocialLinks: chooseFormValue(
+      drafts.visibilitySocialLinks,
+      visibilities.visibilitySocialLinks,
+    ),
+  }),
+);
+
+export const profilePageSelector = createSelector(
+  profileAccountSelector,
+  formValuesSelector,
+  profileImageSelector,
+  saveStateSelector,
+  savePhotoStateSelector,
+  isCurrentUserProfileSelector,
+  draftSocialLinksByPlatformSelector,
+  (
+    account,
+    formValues,
+    profileImage,
+    saveState,
+    savePhotoState,
+    isCurrentUserProfile,
+    draftSocialLinksByPlatform,
+  ) => ({
```

Comment:
Similar to the above comment here - could be DRY'd up further with some `...` operators, but seeing all the data and knowing exactly what's being given to the component has its own merits.